iree/tools/utils/image_util.h

```diff
@@ -0,0 +1,52 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IREE_TOOLS_UTILS_IMAGE_UTIL_H_
+#define IREE_TOOLS_UTILS_IMAGE_UTIL_H_
+
+#include "iree/base/api.h"
+#include "iree/hal/api.h"
+#include "iree/hal/buffer_view.h"
+
+#if __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+// Check the hal buffer information with the image width, height, and channel
+// number. Try to load the image in |filename| with stb_image.h support.
+// Function returns a populated pixel_data buffer and the buffer length.
+// The returned pixel_data buffer must be released by the caller.
```

Comment:
A few comments about the docstring and function signature:

* Reword to focus on what the function does first, then discuss error states and other details.
* Prefix output parameter names with `out_`
* Describe how `|shape|`, `|shape_rank|`, and `|element_type|` are used (not in my suggested comment)
* Move `|filename|` to the first arg, and maybe use `iree_string_view_t`?

```suggestion
// Loads the image at |filename| into |out_pixel_data| and sets
// |out_buffer_length| to it's length.
//
// The file must be in a format supported by stb_image.h.
// The returned |out_pixel_data| buffer must be released by the caller.
```

same for the other functions in this header

Reply:
Done.